packages/strapi-design-system/src/InputJSON/InputJSON.js

```diff
@@ -0,0 +1,127 @@
+import React, { useEffect, useRef, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Field, FieldLabel, FieldError } from '../Field';
+import { Box } from '../Box';
+import { Stack } from '../Stack';
+import { addMarks, filterMarks, lineHighlightMark } from './decorationExtension';
+import * as jsonlint from './utils/jsonLint';
+import { JsonComponent } from './JsonComponent';
+
+const WAIT = 500;
+
+const StyledBox = styled(Box)`
+  border-radius: ${({ theme }) => theme.borderRadius};
+  outline: 1px solid ${({ theme, error }) => (error ? theme.colors.danger600 : 'transparent')};
+`;
+
+export const InputJSON = ({ id, label, value, error, theme, onChange, editable }) => {
+  const editorState = useRef(null);
+  const editorView = useRef(null);
+  const timerRef = useRef();
+
+  const getContentAtLine = (line) => {
+    return editorState.current?.doc?.line(line);
+  };
+
+  const markSelection = useCallback(({ message }) => {
+    let line = parseInt(message.split(':')[0].split('line ')[1], 10);
+    const { text, to: lineEnd } = getContentAtLine(line);
+    const lineStart = lineEnd - text.trimStart().length;
+
+    if (lineEnd > lineStart)
+      editorView.current.dispatch({
+        effects: addMarks.of([lineHighlightMark.range(lineStart, lineEnd)]),
+      });
+  }, []);
+
+  const clearErrorHighlight = () => {
+    const docEnd = editorState.current?.doc?.length || 0;
+    editorView.current.dispatch({
+      effects: filterMarks.of((from, to) => to <= 0 || from >= docEnd),
+    });
+  };
+
+  const validateJSON = useCallback(
+    ({ value, isOnChangeCallback }) => {
```

Comment:
I have added onChange callback to update json object in parent component. This is being done after parsing the json string, which is also called on load. But there is no need to update json object on load, so skipping it with this flag.

Reply:
Okay these sorts of things should be documented in the code though, otherwise someone coming to refactor this in ~1 year, will not know these bits 😄 

Is passing a flag the best way? My concern is `onChange` will probably change it's identity _alot_ in the actual CMS codebase. It might be better to return a value from `validateJSON` and use that return to pass to the `onChange` handler. This means we keep the array dep smaller and more stable and we're not using a flag which i find confusing and muddying the water.